### PR TITLE
be able to use secrets for credentials

### PR DESCRIPTION
**Pull request by jmazzitelli**
A PoC that allows endpoint credentials to refer to a OS secret.

### CODE REVIEW

**Comment by pilhuhn** [comment]
This works for me. I have a secret like this:
`echo -n "test4hawkular" | base64 # create the password`
and then
```
$ less ~/hosa-secret.yml
apiVersion: v1
kind: Secret
metadata:
  name: hosa-secret
type: Opaque
data:
  password: dGVzdDRoYXdrdWxhcg==
```

This is referenced in the `java-exec` container's ENV
```
 containers:
        - env:
          - name: AB_JOLOKIA_PASSWORD_RANDOM
            value: "false"
          - name: AB_JOLOKIA_PASSWORD
            valueFrom:
              secretKeyRef:
                name: hosa-secret
                key: password
```

and also in the config-map for the agent:

```
      credentials:
        username: jolokia
        password: secret:hosa-secret/password
```

**Comment by jmazzitelli** [comment]
I just changed this so it creates our own clusterrole that just gives "get/secrets" permission. It gets assigned to our service account. I tested it and it works.

**Comment by jmazzitelli** [comment]
I took out the granting of the full cluster-reader role. We do not need the vast majority of the permissions cluster-reader provides. So, instead, I create much more limited cluster role and assign only that role to the agent. I tested it and it looks to be working. See the deploy/openshift/hawkular-openshift-agent.yaml for the ClusterRole that is created.

File: Makefile [hunk 1]
```diff
@@ -41,13 +41,14 @@ docker-examples:
...
-	oc adm policy add-cluster-role-to-user cluster-reader system:serviceaccount:openshift-infra:hawkular-openshift-agent
 	oc create -f deploy/openshift/hawkular-openshift-agent-configmap.yaml -n openshift-infra
 	oc process -f deploy/openshift/hawkular-openshift-agent.yaml | oc create -n openshift-infra -f -
+	oc adm policy add-cluster-role-to-user hawkular-openshift-agent system:serviceaccount:openshift-infra:hawkular-openshift-agent
 
 openshift-undeploy:
 	@echo Undeploying the Agent from OpenShift
-	oc delete all,secrets,sa,templates,configmaps,daemonsets --selector=metrics-infra=agent -n openshift-infra
+	oc delete all,secrets,sa,templates,configmaps,daemonsets,clusterroles --selector=metrics-infra=agent -n openshift-infra
+	oc delete clusterroles hawkular-openshift-agent
```
Comment (jmazzitelli) [inline Makefile]: This line that deletes the cluster role can be deleted once this is fixed: https://github.com/openshift/origin/issues/12450
There is already a PR that fixes it, just wait for it to be merged: https://github.com/openshift/origin/pull/12461

**Pull request merged**